compiler/rustc_middle/src/ty/context.rs

```diff
@@ -1039,17 +1039,34 @@ impl<'tcx> TyCtxtAt<'tcx> {
         // This is fine because:
         // - those queries are `eval_always` so we won't miss their result changing;
         // - this write will have happened before these queries are called.
-        let data = def_kind.def_path_data(name);
-        let key = self.untracked.definitions.write().create_def(parent, data);
+        let def_id = self.tcx.create_def(parent, name, def_kind);
 
-        let feed = TyCtxtFeed { tcx: self.tcx, key };
-        feed.def_kind(def_kind);
+        let feed = self.tcx.feed_local_def_id(def_id);
         feed.def_span(self.span);
```

Comment:
Why is this span not simply passed to `create_def`?

Reply:
This span is not available in resolver right now, it will be passed when the FIXME (https://github.com/rust-lang/rust/pull/119136#discussion_r1432708100) is addressed.